### PR TITLE
[Merged by Bors] - chore(LpSpace): cleanup `Fintype`/`Finite`

### PR DESCRIPTION
**Pull request by urkud**
Also rename `*lpBcf` to `*lpBCF` and drop 2 duplicate instances.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by eric-wieser** [comment]
!bench

File: Mathlib/Analysis/NormedSpace/LpEquiv.lean [hunk 1]
```diff
@@ -142,86 +146,93 @@ variable [NormedAddCommGroup E] [NormedSpace 𝕜 E] [NonUnitalNormedRing R]
...
 -- `one_memℓp_infty` to get the `Ring` instance on `lp`.
 /-- The canonical map between `lp (fun _ : α ↦ A) ∞` and `α →ᵇ A` as an `AlgEquiv`. -/
-noncomputable def AlgEquiv.lpBcf : lp (fun _ : α => A) ∞ ≃ₐ[𝕜] α →ᵇ A :=
-  { RingEquiv.lpBcf A with commutes' := fun _k => rfl }
-#align alg_equiv.lp_bcf AlgEquiv.lpBcf
+noncomputable def AlgEquiv.lpBCF : lp (fun _ : α ↦ A) ∞ ≃ₐ[𝕜] α →ᵇ A :=
+  { RingEquiv.lpBCF A with commutes' := fun _k ↦ rfl }
+#align alg_equiv.lp_bcf AlgEquiv.lpBCF
+
+@[deprecated] alias AlgEquiv.lpBcf := AlgEquiv.lpBCF -- 2024/03/16
```
Comment (eric-wieser) [inline Mathlib/Analysis/NormedSpace/LpEquiv.lean:224]:
```suggestion
@[deprecated] alias AlgEquiv.lpBcf := AlgEquiv.lpBCF -- 2024-03-16
```
for iso-8601

Comment (eric-wieser) [reply Mathlib/Analysis/NormedSpace/LpEquiv.lean:224]: (and throughout)

Comment (urkud) [reply Mathlib/Analysis/NormedSpace/LpEquiv.lean]: Done.

File: Mathlib/Analysis/InnerProductSpace/PiL2.lean [hunk 2]
```diff
@@ -151,19 +151,17 @@ theorem EuclideanSpace.sphere_zero_eq {n : Type*} [Fintype n] (r : ℝ) (hr : 0
...
+  inferInstance
 #align euclidean_space.finite_dimensional EuclideanSpace.instFiniteDimensional
 
-instance EuclideanSpace.instInnerProductSpace : InnerProductSpace 𝕜 (EuclideanSpace 𝕜 ι) := by
-  infer_instance
+variable [Fintype ι]
+
+instance EuclideanSpace.instInnerProductSpace : InnerProductSpace 𝕜 (EuclideanSpace 𝕜 ι) :=
+  inferInstance
 #align euclidean_space.inner_product_space EuclideanSpace.instInnerProductSpace
```
Comment (sgouezel) [inline Mathlib/Analysis/InnerProductSpace/PiL2.lean:165]: I'm not sure I get the point of these instances if they can already be derived automatically. Why not just remove them?

**Review by sgouezel — COMMENTED** [review]
bors d+

**Comment by urkud** [comment]
!bench

**Pull request closed without merging**